pydantic/main.py

```diff
@@ -99,6 +121,19 @@ def hash_func(self_: Any) -> int:
 
             cls: type[BaseModel] = super().__new__(mcs, cls_name, bases, namespace, **kwargs)  # type: ignore
 
+            # TODO: cls.__concrete__ has been removed -- do we need to retain this for v1 compatibility?
+            # TODO: cls.__parameters__ may be removed -- do we need to retain this for v1 compatibility?
+            cls.__pydantic_generic_origin__ = __pydantic_generic_origin__
+            cls.__pydantic_generic_args__ = __pydantic_generic_args__
+            cls.__pydantic_generic_parameters__ = __pydantic_generic_parameters__ or getattr(
+                cls, '__parameters__', None
+            )
+            cls.__pydantic_deleted_attrs__ = {}  # TODO: yuck, get rid of this
+            cls.__pydantic_generic_typevars_map__ = (
```

Comment:
Note to self: probably makes sense to remove this attribute since it is computed from the others.

Reply:
I changed my mind as it is reused in a few places. Happy to revisit later.